UnitTests/Fold.cs

```diff
@@ -0,0 +1,138 @@
+            var expected =
+                GenTensor<int, IntWrapper>.CreateTensor(new[,,] {
+                    { { 10101, 20202 }, { 30303, 40404 } },
+                    { { 101010, 202020 }, { 303030, 404040 } },
+                });
+            Assert.AreEqual(expected, acc);
+            Assert.AreEqual(frogCopy, frog);
+        }
+
+        [TestMethod]
+        public void SumAx1()
+        {
+            var acc = GenTensor<int, IntWrapper>.CreateTensor(new TensorShape(3, 2, 2), id => 0);
+            GenTensor<int, IntWrapper>.Fold(new HonkPerf.NET.Core.PureValueDelegate<int, int, int>((a, b) => a + b), acc, frog, axis: 1);
+            var expected =
+                GenTensor<int, IntWrapper>.CreateTensor(new[, ,] {
+                    {
+                        { 11, 22 }, { 33, 44 }
+                    },
+                    {
+                        { 1100, 2200 }, { 3300, 4400 }
+                    },
+                    {
+                        { 110000, 220000 }, { 330000, 440000 }
+                    }
+                });
+            Assert.AreEqual(expected, acc);
+            Assert.AreEqual(frogCopy, frog);
+        }
+
+        [TestMethod]
+        public void SumAx2()
+        {
+            var acc = GenTensor<int, IntWrapper>.CreateTensor(new TensorShape(3, 2, 2), id => 0);
+            GenTensor<int, IntWrapper>.Fold(new HonkPerf.NET.Core.PureValueDelegate<int, int, int>((a, b) => a + b), acc, frog, axis: 2);
+            var expected = 
+                GenTensor<int, IntWrapper>.CreateTensor(new [,,] {
+                    {
+                        { 4, 6 },
+                        { 40, 60 }
+                    },
+                    {
+                        { 400, 600 },
+                        { 4000, 6000 }
+                    },
+                    {
+                        { 40000, 60000 },
+                        { 400000, 600000 }
+                    }
+                });
+            Assert.AreEqual(expected, acc);
+            Assert.AreEqual(frogCopy, frog);
+        }
+
+        [TestMethod]
+        public void SumAx3()
+        {
+            var acc = GenTensor<int, IntWrapper>.CreateTensor(new TensorShape(3, 2, 2), id => 0);
+            GenTensor<int, IntWrapper>.Fold(new HonkPerf.NET.Core.PureValueDelegate<int, int, int>((a, b) => a + b), acc, frog, axis: 3);
+            var expected =
+                GenTensor<int, IntWrapper>.CreateTensor(new[, ,] {
+                    {
+                        { 3, 7 },
+                        { 30, 70 }
+                    },
+                    {
+                        { 300, 700 },
+                        { 3000, 7000 }
+                    },
+                    {
+                        { 30000, 70000 },
+                        { 300000, 700000 }
+                    }
+                });
+            Assert.AreEqual(expected, acc);
+            Assert.AreEqual(frogCopy, frog);
+        }
```

Comment:
Use `[DataTestMethod]` instead of repeating so many times

Reply:
Expected are different

Reply:
Parameterize expected, parameterize axis

Reply:
It's not trivial. Volumes are different. Plus, it won't be as readable

Reply:
ok